kafka/data_source_kafka_topic.go

```diff
@@ -36,23 +39,24 @@ func kafkaTopicDataSource() *schema.Resource {
 	}
 }
 
-func dataSourceTopicRead(d *schema.ResourceData, meta interface{}) error {
+func dataSourceTopicRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	// Unlike the resource topicRead, there is no pre-existing ID. We must use the 'name' to look up the resource.
 	// See https://learn.hashicorp.com/tutorials/terraform/provider-create?in=terraform/providers#implement-read
 	name := d.Get("name").(string)
 
-	client := meta.(*LazyClient)
-	topic, err := client.ReadTopic(name, true)
+	client := meta.(*RetryLazyClient)
+	timeout := 10 * time.Minute
```

Comment:
Terraform only provides timeouts to resources not datasources so using a hard coded 10 minute timeout